pkg/ocm/provider/authorizer/mentix/mentix.go

```diff
@@ -0,0 +1,229 @@
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/cs3org/reva/pkg/ocm/provider"
+	"github.com/cs3org/reva/pkg/ocm/provider/authorizer/registry"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	registry.Register("mentix", New)
+}
+
+// Client is a Mentix API client
+type Client struct {
+	BaseURL    string
+	HTTPClient *http.Client
+}
+
+// New returns a new authorizer object.
+func New(m map[string]interface{}) (provider.Authorizer, error) {
+	c := &config{}
+	if err := mapstructure.Decode(m, c); err != nil {
+		err = errors.Wrap(err, "error decoding conf")
+		return nil, err
+	}
+	c.init()
+
+	client := &Client{
+		BaseURL:    c.URL,
+		HTTPClient: rhttp.GetHTTPClient(context.Background()),
+	}
+
+	return &authorizer{
+		client: client,
+		conf:   c,
+	}, nil
+}
+
+type config struct {
+	URL                   string `mapstructure:"url"`
+	VerifyRequestHostname bool   `mapstructure:"verify_request_hostname"`
+}
+
+func (c *config) init() {
+	if c.URL == "" {
+		c.URL = "http://localhost:9600/mentix"
+	}
+}
+
+type authorizer struct {
+	client *Client
+	conf   *config
+}
+
+func (c *Client) sendRequest(req *http.Request) (*provider.MentixResponse, error) {
+	req.Header.Set("Accept", "application/json; charset=utf-8")
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer res.Body.Close()
+
+	// TODO(mirekys): use providerv1beta1.ListAllProvidersResponse directly after mentix service uses it
+	var mr provider.MentixResponse
+	if err = json.NewDecoder(res.Body).Decode(&mr); err != nil {
+		return nil, err
+	}
+
+	return &mr, nil
+}
+
+func (c *Client) fetchAllProviders() ([]*ocmprovider.ProviderInfo, error) {
+	req, err := http.NewRequest("GET", c.BaseURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.sendRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(mirekys): following translation won't be needed after mentix exporter also uses ocmprovider struct
+	providers := make([]*ocmprovider.ProviderInfo, 0, len(res.Sites))
+	for _, si := range res.Sites {
+		services := make([]*ocmprovider.Service, 0, len(si.Services))
+		for _, se := range si.Services {
+			services = append(services, &ocmprovider.Service{
+				Host: se.Host,
+				Endpoint: &ocmprovider.ServiceEndpoint{
+					Type: &ocmprovider.ServiceType{
+						Name:        se.Type.Name,
+						Description: se.Type.Description,
+					},
+					Name:        se.Name,
+					Path:        se.URL,
+					IsMonitored: se.IsMonitored,
+				},
+			})
+		}
+
+		providers = append(providers, &ocmprovider.ProviderInfo{
+			Name:         si.Name,
+			FullName:     si.FullName,
+			Description:  si.Description,
+			Organization: si.Organization,
+			Domain:       si.Domain,
+			Homepage:     si.Homepage,
+			Email:        si.Email,
+			Services:     services,
+		})
+	}
+
+	status := &rpcv1beta1.Status{
```

Comment:
You can just return providers here. We're not using status anywhere.

Reply:
True indeed, I will get rid of that